SwiftWisdom/Core/StandardLibrary/Numbers/UnsignedInteger+Extensions.swift

```diff
@@ -24,18 +16,48 @@ public func random(inRange range: CountableClosedRange<Int>) -> Int {
     return random
 }
 
-// TODO: write tests for this extension
 extension UnsignedInteger {
     public init(ip_data: NSData) {
         let hexInt = ip_data.ip_hexInt ?? 0
         self.init(ip_safely: hexInt)
     }
-
+    
     public init(ip_data: Data) {
         let hexInt = ip_data.ip_hexInt ?? 0
         self.init(ip_safely: hexInt)
     }
-
+    
+    public init<T : SignedInteger>(ip_safely value: T) {
+        if value < 0 {
+            self.init(ip_safely: UInt8(0))
+        } else {
+            let max = UIntMax(value.toIntMax())
+            self.init(ip_safely: max)
+        }
+    }
+    
+    public init<T : UnsignedInteger>(ip_safely value: T) {
+        self = 0
+        
+        let maxSelf = type(of: self).ip_maxValue
+        if maxSelf.toUIntMax() >= value.toUIntMax() {
+            self = .init(value.toUIntMax())
+        } else {
+            self = maxSelf
+        }
+    }
+    
+    // TODO: write tests for this function
+    public static func ip_random() -> Self {
+        let intMax = Int(ip_maxValue.toIntMax())
+        let rand = random(inRange: 0...intMax)
```

Comment:
add another TODO here, which is to use @andrewdolce's new collection ip_random method when it comes out.